docs/topics/sealed-classes.md

```diff
@@ -12,16 +12,23 @@ In some sense, sealed classes are similar to [`enum`](enum-classes.md) classes:
 for an enum type is also restricted, but each enum constant exists only as a _single instance_, whereas a subclass
 of a sealed class can have _multiple_ instances, each with its own state.
 
+As an example, consider a hierarchy of possible errors in a library. There can be classes for networking, database, and
+other errors, as well as code that uses them, such as handling and logging.
+If a root interface of the error hierarchy is exposed in the public API, client classes can inherit it, creating
+new error types. They are declared outside the library, so it cannot handle them in specific ways. With a sealed hierarchy,
+library authors can be sure that they know all possible error types and no other ones can appear.
```

Comment:
To me, this whole text paragraph is too complicated and inconsequent. Let's try to rewrite it?

Reply:
Tried =) Please check.